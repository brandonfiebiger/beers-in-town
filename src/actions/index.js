export const populateEventsFromLocation = events => ({
    type: 'EVENTS_FROM_LOCATION',
    events
})

export const populateEventsFromSearch = events => ({
  type: 'EVENTS_FROM_SEARCH',
  events
})

export const getPropsFromEvent = eventProps => ({
  type: 'EVENT_TO_VIEW',
  eventProps
})

export const getPropsFromGroup = groupProps => ({
  type: 'GROUP_TO_VIEW',
  groupProps
})

export const populateBreweriesFromLocation = breweries => ({
  type: 'BREWERIES_FROM_LOCATION',
  breweries
})

export const populateBreweriesFromSearch = breweries => ({
  type: 'BREWERIES_FROM_SEARCH',
  breweries
})

export const getLocation = location => ({
  type: 'LOCATION_AVAILABLE',
  location
})

export const populateGroupsFromLocation = groups => ({
  type: 'GROUPS_FROM_LOCATION',
  groups
})

export const populateGroupsFromSearch = groups => ({
  type: 'GROUPS_FROM_SEARCH',
  groups
})

export const hasErrored = error => ({
  type: 'HAS_ERRORED',
  error
})