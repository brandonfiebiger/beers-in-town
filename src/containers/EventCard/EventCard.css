.EventCard {
  border: 2px solid;
  border-radius: 20px; 
  width: 45rem;
  margin-bottom: 2rem;
  height: 17rem;
  color: #574f4f;
  text-shadow: 1px 1px #574f4f;
  box-shadow: 0 0 40px #000000;
  background-image: linear-gradient(to right top, #a5d9bc, #aedab7, #b8dbb2, #c2dcae, #cddcab);
}

h2 {
  text-decoration: underline;
}

.nav {
  color: #574f4f;
  cursor: pointer;
}