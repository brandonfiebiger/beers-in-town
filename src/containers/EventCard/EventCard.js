import React, { Component } from 'react';
import { Route, withRouter, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactHtmlParser, { processNodes, convertNodeToElement, htmlparser2 } from 'react-html-parser';
import  EventInfo  from '../EventInfo/EventInfo';
import { getPropsFromEvent } from '../../actions/index';

class EventCard extends Component {
  constructor(props) {
    super(props);

  }

  handleClick = () => {
    this.props.sendPropsFromCard(this.props)
  }

  render() {
    const { date, desc, group, time, ven, groupName, name, id } = this.props
    return(
    <div>
     <h2>{name}</h2>
     <p>{date}</p>
     <p>{time}</p>
     <h3>{groupName}</h3>
     <h4>{ven.name}</h4>
     <NavLink onClick={this.handleClick} to={`/events/${id}`} className='nav'>event info</NavLink>
    </div>
    )
  }
}

export const mapStateToProps = state => ({
  events: state.events
})

export const mapDispatchToProps = dispatch => ({
  sendPropsFromCard: eventProps => dispatch(getPropsFromEvent(eventProps))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(EventCard));