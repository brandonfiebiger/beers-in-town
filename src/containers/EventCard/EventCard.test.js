import  { EventCard, mapDispatchToProps, mapStateToProps }  from './EventCard';
import { getPropsFromEvent } from '../../actions';
import { shallow, mount } from 'enzyme';
import React from 'react';


describe('EventCard', () => {

  let wrapper;
  let mockDispatch
  let mockFn;
  beforeEach(() => {
    mockFn = jest.fn()
    wrapper = shallow(<EventCard sendPropsFromCard={mockFn}/>)
    mockDispatch = jest.fn()
  })


  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })



    it('should call sendPropsFromCard with the correct parameters', () => {

      const mappedProps = mapDispatchToProps(mockDispatch)

      wrapper.instance().handleClick()

      expect(mockFn).toHaveBeenCalled();
    })
 


  describe('mapStateToProps', () => {

    it('should return an object with the events array', () => {
      const mockState = {
        events: ['some event', 'another event']
      }

      const expected = {
        events: ['some event', 'another event']
      }

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    })
  })

  describe('mapDispatchToProps', () => {
    
    it('calls dispatch with a getPropsFromEvent action', () => {
      const mockDispatch = jest.fn();

      const mockProps = {time: 400, status: 'open'};

      const actionToDispatch = getPropsFromEvent(mockProps);

      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.sendPropsFromCard(mockProps);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})