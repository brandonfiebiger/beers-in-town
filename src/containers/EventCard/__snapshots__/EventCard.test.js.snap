// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventCard
    sendPropsFromCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <p />,
        <p />,
        <h3>
          By: 
</h3>,
        "",
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav"
          onClick={[Function]}
          to="/events/undefined"
>
          event info
</NavLink>,
      ],
      "className": "EventCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "By: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "By: ",
          undefined,
        ],
        "type": "h3",
      },
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "event info",
          "className": "nav",
          "onClick": [Function],
          "to": "/events/undefined",
        },
        "ref": null,
        "rendered": "event info",
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <p />,
          <p />,
          <h3>
            By: 
</h3>,
          "",
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav"
            onClick={[Function]}
            to="/events/undefined"
>
            event info
</NavLink>,
        ],
        "className": "EventCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "By: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "By: ",
            undefined,
          ],
          "type": "h3",
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "event info",
            "className": "nav",
            "onClick": [Function],
            "to": "/events/undefined",
          },
          "ref": null,
          "rendered": "event info",
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
